jupyter_console/interactiveshell.py

```diff
@@ -220,6 +231,47 @@ def handle_execute_reply(self, msg_id, timeout=None):
 
             self.execution_count = int(content["execution_count"] + 1)
 
+    def handle_is_complete_reply(self, msg_id, timeout=None):
+        """
+        Wait for a repsonse from the kernel, and return two values:
+            more? - (boolean) should the frontend ask for more input
+            indent - an indent string to prefix the input
+        Overloaded methods may want to examine the comeplete source. Its is
+        in the self._source_lines_buffered list.
+        """
+        ## Get the is_complete response:
+        msg = None
+        count = 0
+        max_tries = 5
+        while self.client.is_alive() and count < max_tries:
+            try:
+                msg = self.client.shell_channel.get_msg(block=False, timeout=timeout)
+            except Empty:
+                continue
+            if msg["parent_header"].get("msg_id", None) == msg_id:
+                break
+            count += 1
+        ## Handle response:
+        if count == max_tries:
+            warn('The kernel did not respond to an is_complete_request.')
```

Comment:
Is this user visible? I suspect there are enough kernels not yet implementing `is_complete` that it will get annoying to see this every time.